src/folderitemdelegate.cpp

```diff
@@ -178,12 +178,24 @@ void FolderItemDelegate::drawText(QPainter* painter, QStyleOptionViewItemV4& opt
 
   QPalette::ColorGroup cg = opt.state & QStyle::State_Enabled ? QPalette::Normal : QPalette::Disabled;
   if(opt.state & QStyle::State_Selected) {
-    painter->fillRect(selRect, opt.palette.highlight());
+    if(!opt.widget)
+      painter->fillRect(selRect, opt.palette.highlight());
     painter->setPen(opt.palette.color(cg, QPalette::HighlightedText));
   }
   else
     painter->setPen(opt.palette.color(cg, QPalette::Text));
 
+  if (opt.state & QStyle::State_Selected || opt.state & QStyle::State_MouseOver) {
+    if (const QWidget* widget = opt.widget) { // let the style engine do it
```

Comment:
How can it be not true?
You can always assign a value to a pointer, and this will result in 'true' as the condition statement result.

@PCMan Could you please explain the meaning of the following statement?
```c++
if (const QWidget* widget = opt.widget) {
```